zookeeper-server/pom.xml

```diff
@@ -75,8 +75,12 @@
       <artifactId>slf4j-api</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-log4j12</artifactId>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
```

Comment:
These really should be runtime scoped dependencies, since they should not actually be used anywhere in the code at compile time.

Reply:
I agree. @anmolnar is there a reason why these are not in 'test' scope?

Reply:
This one makes me confused: we didn't do this for the log4j1 reference. If you check it in this file, it's not "test" scoped. Also, if I make logback to "test" scoped, it won't be included in the binary distribution. Is that okay?

Reply:
Hmm... you are right. I didn't think this through. We still need to put logback to the binary distribution as we need some default log engine.

Reply:
@anmolnar Yeah, it wasn't that way for log4j1, because log4j1 was used at compile time. However, part of this is to complete the migration to `slf4j-api` at compile time, so the runtime implementations can be chosen by users. So, no actual implementation jar should need to be `compile` scoped anymore. They can just be `runtime` (or `test` if it is needed to compile the tests; `test` implies `runtime` as well). The main point is that they aren't needed at `compile` time, because `slf4j-api` is used at compile time instead.

Reply:
`runtime` dependencies should still be included in the binary distribution... I'd have to check to see how you are constructing the binary distribution. But, you shouldn't put it on the `compile` class path just to put it in the binary distribution. There's better ways to include it in the distribution than changing the scope.